scripts/happy

```diff
@@ -1160,19 +1164,21 @@ def push(ctx, images, tag, extra_tag, compose_env):
     # images to the right places with some re-tagging.
     first_registry = get_ecr_repo(config)
     builder_config["ecr_repo"] = first_registry  # TODO this is a leak in our abstraction, need to fix.
-    # HACK: Sets happy env name to the first extra tag; we need to know the happy env name at docker image build time, unfortunately, since it's not parameterized in the Docker image env
-    rdev_stack_name = extra_tag[0]
-    print(f"Setting happy_env (rdev stack name) to {rdev_stack_name}")
-    builder_config["happy_env"] = rdev_stack_name
+    # Stack name needs to be set in docker image at build time when this build is being triggered by the push-rdev GHA.
+    # Unfortunately, the stack name is not parameterized in the Docker image env and therefore cannot be set in
+    # container at run time.  (Note that images built on localhost via the `single-cell-infra/scripts/happy` script set
+    # this similarly.)
+    if stack_name:
+        print(f"Setting happy_env (rdev stack name) to {stack_name}")
+        builder_config["happy_env"] = stack_name
```

Comment:
Where does `happy_env` get used and does it interfere with `HAPPY_ENV`. Also it looks like `--env` could be used to set `HAPPY_ENV`. If `happy_env` and `HAPPY_ENV` are the same then we could use `--env`.

Reply:
iirc, these are one and the same.   As this code is quite labyrinthine I'm not sure it's worth digging in further--the entire system needs an overhaul. And if we combine repos, all of this complexity will go away.   If there is a simpler change we can make, can you submit a PR? I want to move on... 😄